include/swift/AST/DiagnosticsSema.def

```diff
@@ -117,7 +117,7 @@ ERROR(could_not_use_type_member_on_instance,none,
       "static member %1 cannot be used on instance of type %0",
       (Type, DeclName))
 ERROR(could_not_use_enum_element_on_instance,none,
-      "enum element %0 cannot be referenced as an instance member",
+      "enum case %0 cannot be used as an instance member",
```

Comment:
I also changed _referenced_ to _used_ here. That seems both clearer and more consistent with other diagnostics.

Reply:
This disgnostic should be specialized for implicit `self` so we can offer something more specific than this.  Especially in the common “switch-case is missing a leading dot” examples.

Not something to worry about here, but definitely nice to have in a follow-up.